Fix: switch to `5.1-syntax_5.1` branch for lcf submodule

SlySven opened the pull request:
For the last few years we have been using the wrong branch as the one we were on did not permit the use of `goto` as an identifier and "labels" as targets for such `goto`s as version of Lua after 5.1 forbade that as it became a command to uncontrollably jump around Lua scripts as it does in the C (or BASIC) programming languages!

This change will pull in changes that @keneanung did to the lfc submodule as:
* https://github.com/martin-eden/lua_code_formatter/pull/15
* https://github.com/martin-eden/lua_code_formatter/pull/16
but we had forgotten to use.

This will close #5997.

Signed-off-by: Stephen Lyons <slysven@virginmedia.com>

SlySven committed:
Fix: switch to 5.1-syntax_5.1 branch for lcf submodule

For the last few years we have been using the wrong branch as the one we
were on did not permit the use of `goto` as an identifier and "labels" as
targets for such `goto`s as version of Lua after 5.1 forbode that as it
became a command to uncontrollably jump around Lua scripts as it does in
the C (or BASIC) programming languages!

This change will pull in changes that @keneanung did to the lfc submodule
as:
* https://github.com/martin-eden/lua_code_formatter/pull/15
* https://github.com/martin-eden/lua_code_formatter/pull/16
but we had forgotten to use.

This will close #5997.

Signed-off-by: Stephen Lyons <slysven@virginmedia.com>

vadi2 reviewed (commented):
You also need to have a look at fixing the workflow that auto-updates us to latest version of lcf: https://github.com/Mudlet/Mudlet/blob/development/.github/workflows/update-3rdparty.yml#L134

SlySven commented:
Okay I'll look to change the references from the `5.1` branch to the `5.1-syntax_5.1` one.

BTW I've discovered that there are some errors in that massive chunk of the `./src/mudlet.pro` Qmake project file that is involved in ensuring a `make install` works for the LCF Lua files in a Unix build environment - because I am modifying it to also work for a MSYS2/Minw-w64 `PKGBUILD` one... 🙄 It turns out there are some typos - and with the addition of this PR - also some additions and removals due to the elimination of the handling of `goto` and the labels that that would use in Lua 5.2+ (?).

However for simplicity I will do that in a separate PR once this one has gone in.

I am pleased that I have also found out whilst working on that, that the principles of putting together a package for an OS look likely to be portable to other OSes. I can see better as to how to solve one issue I was having with a FreeBSD "Port" in the same manner. {That is to build the Luarock modules that Mudlet needs, (as binary libraries) and then to have them put in a Unix-ish `/usr/lib/mudlet/` type directory which is the sort of place for an application's own "personal" libraries - so that they won't clash with the installation of the same luarock by the end-user for their own purposes...}

SlySven committed:
Fix: update auto-updater to refer to correct (`5.1-syntax_5.1`) branch

Also add the `./github/` sub-directory files, including the one that needs
editing to perform this change, to the qmake project file - so they show up
in the QtCreator editor...!

Signed-off-by: Stephen Lyons <slysven@virginmedia.com>

SlySven commented:
Ah, that and some other files weren't even listed in the `qmake` project file so I wasn't much aware of them! 😲

vadi2 commented on .github/workflows/update-3rdparty.yml:
```diff
@@ -141,10 +141,10 @@ jobs:
           fetch-depth: 0
 
-      - name: update submodule to latest in 5.1 branch
+      - name: update submodule to latest in 5.1-syntax_5.1 (i.e. only valid for Lua 5.1 scripts) branch
```
I'd keep this as-is, don't need to go into such detail at this level.

SlySven replied:
I think we need to mention it *once* - so that it is clear why we are not using the `5.1` branch instead.

vadi2 replied:
Sure, and it is mentioned in other places already. Don't need to do it here as well.

SlySven replied:
Where, precisely?

SlySven replied:
Where, precisely?

vadi2 replied:
![image](https://user-images.githubusercontent.com/110988/156841798-0f0730ef-d72e-45fb-a06b-fe79367ea315.png)

vadi2 replied:
The problem with using the `name` for it is that it's not meant to be a long one:

![image](https://user-images.githubusercontent.com/110988/156841915-d542a566-3cba-4bc5-921a-a7ba70bedadc.png)

SlySven replied:
Okay, somehow whatever was sticking for me on GitHub seems to have unglued itself.

So what is the problem? Is it that there are now four places where I have uses `5.1-syntax_5.1` instead of `5.1` - that you have pictured - because that is the branch that is being used - or the single case (the second of those four) where I have given a full explanation? I want to avoid saying the `5.1` branch because that is explicitly NOT the branch we want.

If we really must keep the texts short then I would drop most of the references to `5.1` except the `git checkout` one (obviously) but leave the long explanation one that will appear in the body of the auto generated PR/commit message.

Your thoughts, please?

SlySven replied:
Okay, somehow whatever was sticking for me on GitHub seems to have unglued itself.

So what is the problem? Is it that there are now four places where I have uses `5.1-syntax_5.1` instead of `5.1` - that you have pictured - because that is the branch that is being used - or the single case (the second of those four) where I have given a full explanation? I want to avoid saying the `5.1` branch because that is explicitly NOT the branch we want.

If we really must keep the texts short then I would drop most of the references to `5.1` except the `git checkout` one (obviously) but leave the long explanation one that will appear in the body of the auto generated PR/commit message.

Your thoughts, please?

SlySven replied:
Okay, somehow whatever was sticking for me on GitHub seems to have unglued itself.

So what is the problem? Is it that there are now four places where I have uses `5.1-syntax_5.1` instead of `5.1` - that you have pictured - because that is the branch that is being used - or the single case (the second of those four) where I have given a full explanation? I want to avoid saying the `5.1` branch because that is explicitly NOT the branch we want.

If we really must keep the texts short then I would drop most of the references to `5.1` except the `git checkout` one (obviously) but leave the long explanation one that will appear in the body of the auto generated PR/commit message.

Your thoughts, please?

SlySven replied:
Okay, somehow whatever was sticking for me on GitHub seems to have unglued itself.

So what is the problem? Is it that there are now four places where I have uses `5.1-syntax_5.1` instead of `5.1` - that you have pictured - because that is the branch that is being used - or the single case (the second of those four) where I have given a full explanation? I want to avoid saying the `5.1` branch because that is explicitly NOT the branch we want.

If we really must keep the texts short then I would drop most of the references to `5.1` except the `git checkout` one (obviously) but leave the long explanation one that will appear in the body of the auto generated PR/commit message.

Your thoughts, please?

SlySven replied:
Can you delete those duplicates - I can't as the drop down menu doesn't appear any more (like previews or the posted messages neither :roll_eyes:)

vadi2 replied:
The `- name:` is not what will be in the body of the PR, you are across that right?

vadi2 commented on src/mudlet.pro:
```diff
@@ -1579,6 +1579,19 @@ unix:!macx {
 
 
 DISTFILES += \
+    ../.github/CODEOWNERS \
```
Won't this install these files with `make install`? These files don't need to be on the users computer

SlySven replied:
Ah, yeah, it should be the `OTHER_FILES` variable - strangely I can't see that documented currently...

SlySven committed:
Fix: correct the QMake variable used to record some project files

In the previous commit I added some files to the `DISTFILES` qmake variable
but that is for files that should be included in a `make dist` (source code
tarball) Unix makefile usage. Given that the `.github` directory files are
not ones that would be used in this manner (they are only used by the
Mudlet project to run it's own CI/CB process) they should instead be
assigned to the (undocumented in the qmake documentation) `OTHER_FILES`
variable.

There are further changes needed to the QMake `OTHER_FILES`, `INSTALLS` and
`DISTFILES` variables but they should be left to a further, separate PR.

Signed-off-by: Stephen Lyons <slysven@virginmedia.com>

SlySven commented:
:boom: Sorry, but I am having major borkage with GitHub and my web-browser that is not a *Chrome-clone*. Something to do with Micro$oft and G@@gle trying to make things shiny on top but introducing :poop: underneath the surface.

https://user-images.githubusercontent.com/340855/153508596-f05c2aa6-9585-4505-aa94-b6a66143bc5d.png

:scream:

SlySven committed:
Revise: improve naming of item

Shorten it in three of four places to not mention "5.1" but ensure the
generated commit message says why we use the specific branch.

Signed-off-by: Stephen Lyons <slysven@virginmedia.com>

SlySven committed:
Revise: trim a comment

I was just trying to remember what `DISTFILES` was for...!

Signed-off-by: Stephen Lyons <slysven@virginmedia.com>
